src/thunderbots/software/ai/navigator/path_planner/path_planner.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+#include <ai/world/world.h>
+
+#include <optional>
+#include <vector>
+
+#include "geom/point.h"
+
+class PathPlanner
+{
+   public:
+    // TODO: consider passing a lambda to the path planner instead of the world
+    virtual std::optional<std::vector<Point>> findPath(const World &world,
```

Comment:
what is the world for here?

Reply:
A* needs some kind of information on where obstacles are, I was considering changing this to accept a lambda that consumes a `Point` and returns the violation distance.

Reply:
I've changed the function signature to consume a `ViolationFunction`, which is a lambda that consumes a `Point` and returns violation distance. This way, the navigator still has responsibility for what constitutes a violation.

Reply:
What is the violation distance exactly? Is that equivalent to the `g` score in the `f = g + h` for A*?

Reply:
violation distance is an additional term in both the `cost` of an edge and the heuristic that represents how much we are intruding into the bounds of an obstacle.